src/restoreTerminals.ts

```diff
@@ -63,8 +68,15 @@ export default async function restoreTerminals(configuration: Configuration) {
         );
       }
     } else {
+      let color = TerminalColor.white;
+
+      if (terminalWindow.splitTerminals[0]?.color) {
+        color = TerminalColor[terminalWindow.splitTerminals[0].color];
+      }
+
       term = vscode.window.createTerminal({
         name: name,
+        color: new vscode.ThemeColor(color),
```

Comment:
I'm not sure what happens if you call `vscode.ThemeColor` constructor with undefined, but it's better to ensure that `color` never is white instead of undefined.